src/code/z_actor.c

```diff
@@ -2707,19 +2708,88 @@ void Actor_DrawLensActors(PlayState* play, s32 numInvisibleActors, Actor** invis
     CLOSE_DISPS(gfxCtx, "../z_actor.c", 6284);
 }
 
-s32 func_800314B0(PlayState* play, Actor* actor) {
-    return func_800314D4(play, actor, &actor->projectedPos, actor->projectedW);
+/**
+ * Checks if an actor should be culled or not, by seeing if it is contained within its own culling volume.
+ * For more details on the culling test, see `Actor_CullingVolumeTest`.
+ *
+ * "Culling" in this context refers to the removal of something for the sake of improving performance.
+ * For actors, being culled means that their Update and Draw processes are halted.
+ * While halted, an Actor's update state is frozen and it will not draw, making it invisible.
+ *
+ * Actors that are within the bounds of their culling volume may update and draw, while actors that are
+ * out of bounds of its culling volume may be excluded from updating and drawing until they are within bounds.
+ *
+ * It is possible for actors to opt out of update culling or draw culling.
+ * This is set per-actor with `ACTOR_FLAG_UPDATE_CULLING_DISABLED` and `ACTOR_FLAG_DRAW_CULLING_DISABLED`.
+ *
+ * Note: Even if either `ACTOR_FLAG_UPDATE_CULLING_DISABLED` or `ACTOR_FLAG_DRAW_CULLING_DISABLED` are set, the actor
+ * will still undergo the culling test and set `ACTOR_FLAG_INSIDE_CULLING_VOLUME` accordingly.
+ * So, `ACTOR_FLAG_INSIDE_CULLING_VOLUME` cannot be used on it own to determine if an actor is actually culled.
+ * It simply says whether or not they are physically located within the bounds of the culling volume.
```

Comment:
Comment what the return value means? Does it return true if it's within the bounds, or true if it's culled?

Maybe naming it something like `Actor_IsInsideCullingVolume` would help too. Ditto with `Actor_CullingVolumeTest`, although both functions could be called `Actor_IsInsideCullingVolume` so idk how to differentiate them. It's also not very important because these functions are used like twice

Reply:
Yeah the whole wrapping thing is annoying. 

I considered `Actor_IsInsideCullingVolume` as a name, but had the same consideration as you. Which deserves it?

`Actor_CullingCheck` is the high level function which will pass in the actors projected position for you. 
Meanwhile `Actor_CullingVolumeTest` lets you pass the arguments used for testing. 
Epona seems to call this function herself and passes projected values using `SkinMatrix_Vec3fMtxFMultXYZW`. I think this effectivley ends up being the same thing? This mathy out of my wheelhouse though.

Anyway, I would prefer my current names, with the function comment explaining the return (which I will add)

Reply:
What about:

`Actor_CullingCheck` -> `Actor_IsInsideCullingVolume` (checks if the actor itself is inside the culling volume)
`Actor_CullingVolumeTest` -> `Actor_IsPointInsideCullingVolume` (checks if a generic point is inside the actor's culling volume)

Reply:
I dont like the names because I have become attached to what I named them, and like them more purely based on vibes (lol)
I will warm up to your suggestions and likely make the change soon :D 

To try to be more concrete, I simply like that the call used to check culling for actors generally is just "culling check" and that the implementation of the culling volume test is named as such. It is less wordy and feels easier to understand from a more high level, simplistic view

Reply:
`Actor_CullingCheck` is ambiguous IMO, one could expect it to return true if the actor is culled, which is the exact opposite of what it does, so that's why I wanted to put the "direction" in the name. This function is used exactly once though so I don't care too much about it

Reply:
In my view its still going to be confusing and require further reading, because "Inside Culling Volume" still doesnt tell you directly the status of culling (because unintuitively, being outside the culling volume is what means culling occurs)

The 1 usage you mention does make it clear which return corresponds to inside/outside at least.
But either way it will require reading the full function comment to grasp what is happening
```c
                if (Actor_CullingCheck(play, actor)) {
                    actor->flags |= ACTOR_FLAG_INSIDE_CULLING_VOLUME;
                } else {
                    actor->flags &= ~ACTOR_FLAG_INSIDE_CULLING_VOLUME;
                }
```
Like I say I will probably warm up to your suggestion anyway, just trying to explain my pov

Reply:
Ive added the comment on the return. Ill leave the names for now, and encourage other people reviewing to weigh in here.

Reply:
The names for those two functions are fine given the weirdness of the system. I think expecting people to read the writeup about how this works is reasonable.